### PR TITLE
Computer Vision Quickstarts Moved to v5 of SDK

### PR DESCRIPTION
**Pull request by harishkrishnav**
## Purpose
Adding a few quickstarts

### CODE REVIEW

**Comment by harishkrishnav** [comment]
This is probably the last PR for adding quickstarts in this repo in the near future.

File: samples/ComputerVision/DomainSpecificContent/Program.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            string localImagePath = @"Images\celebrities.jpg";   // See this repo's readme.md for info on how to get these images. Alternatively, you can just set the path to any appropriate image on your machine.
+            string remoteImageUrl = "https://github.com/Azure-Samples/cognitive-services-sample-data-files/raw/master/ComputerVision/Images/landmark.jpg";
+
+            Console.WriteLine("Domain Specific Content being analyzed ...");
+            await DomainSpecificContentFromUrlAsync(computerVision, remoteImageUrl);
+            await DomainSpecificContentFromStreamAsync(computerVision, localImagePath);
+        }
+
+        // Analyze a remote image
+        private static async Task DomainSpecificContentFromUrlAsync(ComputerVisionClient computerVision, string imageUrl)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/DomainSpecificContent/Program.cs]: Add a required string parameter for the domain name, so the caller can pass it in and turn this into a more generic method.

File: samples/ComputerVision/DomainSpecificContent/Program.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            
+            DomainModelResults analysis = await computerVision.AnalyzeImageByDomainAsync("landmarks",imageUrl);  //change the first parameter to "landmarks" if that is the domain you are interested in
+
+            Console.WriteLine(imageUrl);
+            DisplayResults(analysis); 
+        }
+
+        // Analyze a local image
+        private static async Task DomainSpecificContentFromStreamAsync(ComputerVisionClient computerVision, string imagePath)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/DomainSpecificContent/Program.cs]: Add a required string parameter for the domain name, so the caller can pass it in and turn this into a more generic method.

File: samples/ComputerVision/ExtractText/Program.cs [hunk 3]
```diff
@@ -42,12 +42,12 @@ public static async Task RunAsync(string endpoint, string key)
...
             Console.WriteLine("Text being extracted ...");
-            await ExtractTextFromStreamAsync(computerVision, localImagePath, numberOfCharsInOperationId, TextRecognitionMode.Handwritten); // if localImagePath points to an image with printed text, use TextRecognitionMode.Printed 
-            await ExtractTextFromUrlAsync(computerVision, remoteImageUrl, numberOfCharsInOperationId, TextRecognitionMode.Printed);
+            await ExtractTextFromStreamAsync(computerVision, localImagePath, numberOfCharsInOperationId); 
+            await ExtractTextFromUrlAsync(computerVision, remoteImageUrl, numberOfCharsInOperationId);
         }
 
         // Read text from a remote image
-        private static async Task ExtractTextFromUrlAsync(ComputerVisionClient computerVision, string imageUrl, int numberOfCharsInOperationId, TextRecognitionMode textRecognitionMode)
+        private static async Task ExtractTextFromUrlAsync(ComputerVisionClient computerVision, string imageUrl, int numberOfCharsInOperationId)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/ExtractText/Program.cs]: Rename to BatchReadFileFromUrlAsync. Please also rename the name of the project and the directory name to BatchReadFile.

File: samples/ComputerVision/ExtractText/Program.cs [hunk 4]
```diff
@@ -56,12 +56,12 @@ private static async Task ExtractTextFromUrlAsync(ComputerVisionClient computerV
...
 
             // Start the async process to read the text
-            BatchReadFileHeaders textHeaders = await computerVision.BatchReadFileAsync(imageUrl, textRecognitionMode);
+            BatchReadFileHeaders textHeaders = await computerVision.BatchReadFileAsync(imageUrl);
             await GetTextAsync(computerVision, textHeaders.OperationLocation, numberOfCharsInOperationId);
         }
 
         // Recognize text from a local image
-        private static async Task ExtractTextFromStreamAsync(ComputerVisionClient computerVision, string imagePath, int numberOfCharsInOperationId, TextRecognitionMode textRecognitionMode)
+        private static async Task ExtractTextFromStreamAsync(ComputerVisionClient computerVision, string imagePath, int numberOfCharsInOperationId)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/ExtractText/Program.cs]: Rename to BatchReadFileFromStreamAsync.

File: samples/ComputerVision/RecognizeText/Program.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            {
+                Endpoint = endpoint
+            };
+            const int numberOfCharsInOperationId = 36;               //This is not a tune-able hyperparameter
+
+            string localImagePath = @"Images\handwritten_text.jpg";  // See this repo's readme.md for info on how to get these images. Alternatively, you can just set the path to any appropriate image on your machine.
+            string remoteImageUrl = "https://github.com/Azure-Samples/cognitive-services-sample-data-files/raw/master/ComputerVision/Images/printed_text.jpg";
+
+            Console.WriteLine("Text being recognized ...");
+            await RecognizeTextFromStreamAsync(computerVision, localImagePath, numberOfCharsInOperationId, TextRecognitionMode.Handwritten);  //Use TextRecognitionMode.Printed for printed text
```
Comment (CarlosP-MS) [inline samples/ComputerVision/RecognizeText/Program.cs]: I don't think the text recognition mode is needed anymore

File: samples/ComputerVision/RegionOfInterest/Program.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+
+            Console.WriteLine("\nPress ENTER to exit.");
+            Console.ReadLine();
+        }
+    }
+    public class areaOfInterestSample
```
Comment (CarlosP-MS) [inline samples/ComputerVision/RegionOfInterest/Program.cs]: AreaOfInterestSample. Please also rename the folder to AreaOfInterest so we are consistent.

File: samples/ComputerVision/RegionOfInterest/Program.cs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            string localImagePath = @"Images\objects.jpg";   // See this repo's readme.md for info on how to get these images. Alternatively, you can just set the path to any appropriate image on your machine.
+            string remoteImageUrl = "https://github.com/Azure-Samples/cognitive-services-sample-data-files/raw/master/ComputerVision/Images/faces.jpg";
+
+            Console.WriteLine("area of interest being found ...");
+            await RoiFromUrlAsync(computerVision, remoteImageUrl);
+            await RoiFromStreamAsync(computerVision, localImagePath);
+        }
+
+        // Analyze a remote image
+        private static async Task RoiFromUrlAsync(ComputerVisionClient computerVision, string imageUrl)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/RegionOfInterest/Program.cs]: GetAreaOfInterestFromUrlAsync

File: samples/ComputerVision/RegionOfInterest/Program.cs [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+            }
+
+            using (Stream imageStream = File.OpenRead(imagePath))
+            {
+                AreaOfInterestResult analysis = await computerVision.GetAreaOfInterestInStreamAsync(imageStream);
+                Console.WriteLine(imagePath);
+                Displayareas(analysis);
+            }
+        }
+        private static void Displayareas(AreaOfInterestResult analysis)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/RegionOfInterest/Program.cs]: DisplayAreas

File: samples/ComputerVision/readme.md [hunk 9]
```diff
@@ -18,6 +18,10 @@ The projects in this code are quickstarts that show how to use Computer Vision S
 | DetectObjects | Detects objects in an image |
 | ExtractText | Looks for printed or handwritten text in an image|
 | OCR | Performs OCR on the image|
+| RegionOfInterest | Gets area of interest|
```
Comment (CarlosP-MS) [inline samples/ComputerVision/readme.md]: AreaOfInterest

File: samples/ComputerVision/readme.md [hunk 10]
```diff
@@ -18,6 +18,10 @@ The projects in this code are quickstarts that show how to use Computer Vision S
 | DetectObjects | Detects objects in an image |
 | ExtractText | Looks for printed or handwritten text in an image|
```
Comment (CarlosP-MS) [inline samples/ComputerVision/readme.md]: BatchReadFile

File: samples/ComputerVision/AreaOfInterest/Program.cs [hunk 11]
```diff
@@ -41,12 +41,12 @@ public static async Task RunAsync(string endpoint, string key)
...
             Console.WriteLine("area of interest being found ...");
-            await RoiFromUrlAsync(computerVision, remoteImageUrl);
-            await RoiFromStreamAsync(computerVision, localImagePath);
+            await AreaOfInterestFromUrlAsync(computerVision, remoteImageUrl);
+            await AreaOfInterestFromStreamAsync(computerVision, localImagePath);
         }
 
         // Analyze a remote image
-        private static async Task RoiFromUrlAsync(ComputerVisionClient computerVision, string imageUrl)
+        private static async Task AreaOfInterestFromUrlAsync(ComputerVisionClient computerVision, string imageUrl)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/AreaOfInterest/Program.cs]: Let's make it consistent with the API name, and call it GetAreaOfInterestFromUrlAsync

File: samples/ComputerVision/AreaOfInterest/Program.cs [hunk 12]
```diff
@@ -57,11 +57,11 @@ private static async Task RoiFromUrlAsync(ComputerVisionClient computerVision, s
             AreaOfInterestResult analysis = await computerVision.GetAreaOfInterestAsync(imageUrl);
 
             Console.WriteLine(imageUrl);
-            Displayareas(analysis);
+            DisplayAreas(analysis);
         }
 
         // Analyze a local image
-        private static async Task RoiFromStreamAsync(ComputerVisionClient computerVision, string imagePath)
+        private static async Task AreaOfInterestFromStreamAsync(ComputerVisionClient computerVision, string imagePath)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/AreaOfInterest/Program.cs]: GetAreaOfInterestFromStreamAsync

File: samples/ComputerVision/AreaOfInterest/Program.cs [hunk 13]
```diff
@@ -73,10 +73,10 @@ private static async Task RoiFromStreamAsync(ComputerVisionClient computerVision
             {
                 AreaOfInterestResult analysis = await computerVision.GetAreaOfInterestInStreamAsync(imageStream);
                 Console.WriteLine(imagePath);
-                Displayareas(analysis);
+                DisplayAreas(analysis);
             }
         }
-        private static void Displayareas(AreaOfInterestResult analysis)
+        private static void DisplayAreas(AreaOfInterestResult analysis)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/AreaOfInterest/Program.cs]: DisplayAreaOfInterest, so it is more clear and also indicates that there is only 1 area, not several

File: samples/ComputerVision/DomainSpecificContent/Program.cs [hunk 14]
```diff
@@ -41,27 +41,27 @@ public static async Task RunAsync(string endpoint, string key)
...
             Console.WriteLine("Domain Specific Content being analyzed ...");
-            await DomainSpecificContentFromUrlAsync(computerVision, remoteImageUrl);
-            await DomainSpecificContentFromStreamAsync(computerVision, localImagePath);
+            await DomainSpecificContentFromUrlAsync(computerVision, remoteImageUrl, "landmarks");
+            await DomainSpecificContentFromStreamAsync(computerVision, localImagePath, "celebrities");
         }
 
         // Analyze a remote image
-        private static async Task DomainSpecificContentFromUrlAsync(ComputerVisionClient computerVision, string imageUrl)
+        private static async Task DomainSpecificContentFromUrlAsync(ComputerVisionClient computerVision, string imageUrl, string specificDomain)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/DomainSpecificContent/Program.cs]: RecognizeDomainSpecificContentFromUrlAsync would be a better name (method names work better when they start with a verb). Please also change the folder/project name to reflect that.

File: samples/ComputerVision/DomainSpecificContent/Program.cs [hunk 15]
```diff
@@ -41,27 +41,27 @@ public static async Task RunAsync(string endpoint, string key)
...
         {
             if (!Uri.IsWellFormedUriString(imageUrl, UriKind.Absolute))
             {
                 Console.WriteLine("\nInvalid remote image url:\n{0} \n", imageUrl);
                 return;
             }
             
-            DomainModelResults analysis = await computerVision.AnalyzeImageByDomainAsync("landmarks",imageUrl);  //change the first parameter to "landmarks" if that is the domain you are interested in
+            DomainModelResults analysis = await computerVision.AnalyzeImageByDomainAsync(specificDomain,imageUrl);  //change the first parameter to "landmarks" if that is the domain you are interested in
```
Comment (CarlosP-MS) [inline samples/ComputerVision/DomainSpecificContent/Program.cs]: please add a space between the two parameters

File: samples/ComputerVision/DomainSpecificContent/Program.cs [hunk 16]
```diff
@@ -41,27 +41,27 @@ public static async Task RunAsync(string endpoint, string key)
...
 
             Console.WriteLine(imageUrl);
             DisplayResults(analysis); 
         }
 
         // Analyze a local image
-        private static async Task DomainSpecificContentFromStreamAsync(ComputerVisionClient computerVision, string imagePath)
+        private static async Task DomainSpecificContentFromStreamAsync(ComputerVisionClient computerVision, string imagePath, string specificDomain)
```
Comment (CarlosP-MS) [inline samples/ComputerVision/DomainSpecificContent/Program.cs]: RecognizeDomainSpecificContentFromStreamAsync

File: samples/ComputerVision/AreaOfInterest/Program.cs [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+            {
+                AreaOfInterestResult analysis = await computerVision.GetAreaOfInterestInStreamAsync(imageStream);
+                Console.WriteLine(imagePath);
+                DisplayAreas(analysis);
+            }
+        }
+        private static void DisplayAreas(AreaOfInterestResult analysis)
+        {
+            Console.WriteLine("The Area of Interest is at location {0},{1},{2},{3}",
+                analysis.AreaOfInterest.X, analysis.AreaOfInterest.W,
```
Comment (CarlosP-MS) [inline samples/ComputerVision/AreaOfInterest/Program.cs]: We should be consistent with how we display regions in the other samples. Please use X+W and Y+H for the data being printed out.

**Pull request merged**